cloud2cloud-gateway/service/service.go

```diff
@@ -38,97 +38,89 @@ var authRules = map[string][]kitNetHttp.AuthArgs{
-				regexp.MustCompile(`r:resources:.*`),
+				regexp.MustCompile(`r:.*`),
 			},
 		},
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}[\/]*$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`r:deviceinformation:.*`),
+				regexp.MustCompile(`r:.*`),
 			},
 		},
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}[\/]*\?` + ContentQuery + `=` + ContentQueryBaseValue + `$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`r:deviceinformation:.*`),
+				regexp.MustCompile(`r:.*`),
 			},
 		},
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}[\/]*\?` + ContentQuery + `=` + ContentQueryAllValue + `$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`r:deviceinformation:.*`),
-				regexp.MustCompile(`r:resources:.*`),
+				regexp.MustCompile(`r:.*`),
 			},
 		},
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}[\/]+.*[\/]*$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`r:resources:.*`),
+				regexp.MustCompile(`r:.*`),
 			},
 		},
 	},
 	http.MethodPost: {
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+subscriptions[\/]*$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`r:deviceinformation:.*`),
-				regexp.MustCompile(`w:subscriptions:.*`),
+				regexp.MustCompile(`r:.*`),
 			},
 		},
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}[\/]+subscriptions[\/]*$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`r:deviceinformation:.*`),
-				regexp.MustCompile(`w:subscriptions:.*`),
+				regexp.MustCompile(`r:.*`),
 			},
 		},
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}[\/]+.*[\/]+subscriptions[\/]*$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`r:resources:.*`),
-				regexp.MustCompile(`w:subscriptions:.*`),
+				regexp.MustCompile(`r:.*`),
 			},
 		},
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}[\/]+.*[\/]*$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`w:resources:.*`),
+				regexp.MustCompile(`w:.*`),
 			},
 		},
 	},
 	http.MethodDelete: {
 		{
 			URI: regexp.MustCompile(`[\/]+api[\/]+v1[\/]+devices[\/]+subscriptions[\/]+[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}[\/]*$`),
 			Scopes: []*regexp.Regexp{
-				regexp.MustCompile(`r:deviceinformation:.*`),
-				regexp.MustCompile(`w:subscriptions:.*`),
+				regexp.MustCompile(`r:.*`),
```

Comment:
I'm not yet aware of what do these scopes mean, but the fact that `Delete` doesn't require `write` seems weird to me. Is it that OK?

Reply:
By spec https://openconnectivity.org/specs/OCF_Cloud_API_For_Cloud_Services_Specification_v2.2.4.pdf it is OK.

Reply:
@Danielius1922 

See below answer of Mart Trayer about read scope of subscription POST/DELETE APIs from email thread discussion: 
> Greetings,
>
>You’re conflating the HTTP method used with what actually happens to the underlying data.
>
>A subscription for events is akin to an observe in CoAP (and other protocols), it doesn’t change any underlying data, it simply reports on changes triggered by others.  You don’t need to have update privileges (or scope) in order to be able to retrieve information/data.  The POST/DELETE is creating/removing the subscription, it is not changing the data that is being notified, hence “r:*” is perfectly acceptable and reasonable. 
>
>Best,
>mark.

Reply:
@Trishia, I see, thanks for the explanation.